### PR TITLE
[feat] Remove `sched_*` internal variables

### LINKED ISSUE #1368: Remove `sched_*` internal variables

### PR DESCRIPTION
**Pull request by rsarm**
Closes #1368

### CODE REVIEW

File: unittests/test_schedulers.py [hunk 1]
```diff
@@ -188,7 +178,7 @@ def _expected_pbs_directives(job):
         ':mem=100GB:cpu_type=haswell' % (num_nodes,
                                          job.num_tasks_per_node,
                                          num_cpus_per_node),
-        '#PBS -q %s' % job.sched_partition,
+        # '#PBS -q %s' % job.sched_partition,
```
Comment (ekouts) [inline unittests/test_schedulers.py:181]: You probably wanted to remove this line, not just comment it

File: reframe/frontend/cli.py [hunk 2]
```diff
@@ -595,28 +595,33 @@ def print_infoline(param, value):
 
         options.flex_alloc_nodes = options.flex_alloc_nodes or 'idle'
         if options.account:
+            options.job_options.append(f'account={options.account}')
             printer.warning(f"`--account' is deprecated and "
                             f"will be removed in the future; you should "
                             f"use `-J account={options.account}'")
 
         if options.partition:
+            options.job_options.append(f'partition={options.partition}')
```
Comment (ekouts) [inline reframe/frontend/cli.py:604]: @vkarak do we want to keep the option working only for slurm? Is there a way to pass this argument for both pbs and slurm through `job_options`?

File: unittests/test_schedulers.py [hunk 3]
```diff
@@ -781,30 +757,13 @@ def test_flex_alloc_invalid_constraint(make_flexible_job):
         prepare_job(job)
 
 
-def test_flex_alloc_valid_reservation_cmd(make_flexible_job):
-    job = make_flexible_job('all',
-                            sched_access=['--constraint=f2'],
-                            sched_reservation='dummy')
```
Comment (vkarak) [inline unittests/test_schedulers.py:787]: This test should remain except that the reservation should be passed through `sched_options`.

File: unittests/test_schedulers.py [hunk 4]
```diff
@@ -781,30 +757,13 @@ def test_flex_alloc_invalid_constraint(make_flexible_job):
...
     job = make_flexible_job('all', sched_access=['--constraint=f2'])
     job.options = ['--reservation=dummy']
     prepare_job(job)
     assert job.num_tasks == 4
 
 
-def test_flex_alloc_exclude_nodes_cmd(make_flexible_job):
-    job = make_flexible_job('all',
-                            sched_access=['--constraint=f1'],
-                            sched_exclude_nodelist='nid00001')
```
Comment (vkarak) [inline unittests/test_schedulers.py:803]: Same comment as above.

**Comment by rsarm** [comment]
After discussing with @vkarak we are postponing this Reframe 3.2 when we will remove `sched_partition` as well.

**Comment by rsarm** [comment]
Closing in favor of https://github.com/eth-cscs/reframe/pull/1483

**Pull request closed without merging**